nautobot_device_lifecycle_mgmt/metrics.py

```diff
@@ -0,0 +1,140 @@
+"""Nautobot Lifecycle Management plugin application level metrics ."""
+from prometheus_client.core import GaugeMetricFamily
+
+from nautobot_device_lifecycle_mgmt.models import DeviceSoftwareValidationResult, InventoryItemSoftwareValidationResult
+
+
+def metrics_lcm_validation_report_device_type():
+    """Calculate number of devices with valid software by device_type.
+
+    Yields:
+        GaugeMetricFamily: Prometheus Metrics
+    """
+    device_validation_model = (
+        DeviceSoftwareValidationResult.objects.values_list("device__device_type__model", flat=True)
+        .distinct()
+        .order_by()
+    )
+
+    valid_device_software_gauge = GaugeMetricFamily(
+        "nautobot_lcm_valid_by_device_type_total",
+        "Number of devices that have valid software by device_type",
+        labels=["device_type"],
+    )
+
+    invalid_device_software_gauge = GaugeMetricFamily(
+        "nautobot_lcm_invalid_by_device_type_total",
+        "Number of devices that have invalid software by device_type",
+        labels=["device_type"],
+    )
+
+    # If there is available data a metric gauge will be created if not no gauge will be created.
+    for model in device_validation_model:
+        valid_device_software_gauge.add_metric(
+            labels=[model],
+            value=(
+                DeviceSoftwareValidationResult.objects.filter(
+                    device__device_type__model=model, is_validated=True
+                ).count()
+            ),
+        )
+        invalid_device_software_gauge.add_metric(
+            labels=[model],
+            value=(
+                DeviceSoftwareValidationResult.objects.filter(
+                    device__device_type__model=model, is_validated=False
+                ).count()
+            ),
+        )
```

Comment:
> if not no gauge will be created

According to [best practices](https://prometheus.io/docs/practices/instrumentation/#avoid-missing-metrics) you should _always_ provide a metric, even if the sample is just set to 0 and no labels are present.